src/shared/install.c

```diff
@@ -1056,11 +1058,13 @@ static int config_parse_also(
                 if (r == 0)
                         break;
 
-                r = install_info_add(c, word, NULL, NULL);
+                r = install_info_add(c, word, NULL, &alsoinfo);
                 if (r < 0)
                         return r;
+                if (r == 1) /* new UnitFileInstallInfo */
+                        alsoinfo->auxiliary = true;
```

Comment:
hmm, this logic would break if we operate on multiple unit files at once, when one of the unit file references another one of the set via Also=, no?

I mean, I think this would have to be inverted: mark the install info structures that are "main", and then treat all unmarked ones as "auxiliary". Only then we can properly handle the case where we shall operate on units A, B, where A also contains Also=B unbeknownst to the user, or am I missing somehting?